openlibrary/macros/UserMetadata.html

```diff
@@ -0,0 +1,124 @@
+$def with (work, edition)
+
+$ username = ctx.user and ctx.user.key.split('/')[-1]
+$ has_edition = edition is not None
+
+<div class="cta-section" style="text-align:center; margin-top:7px;">
+  <a id="modal-link" href="javascript:;">What did you think?</a>
+</div>
+
+<div class="hidden">
+  <div id="metadata-form" class="floaterAdd">
+    <div class="floaterHead">
+      <h2>Share your thoughts</h2>
+      <a class="dialog--close">&times;<span class="shift">$_("Close")</span></a>
+    </div>
+    <form method="post" class="floatform" name="user-metadata" id="user-metadata"></form>
+  </div>
+</div>
+
+<script type="text/javascript">
+window.q.push(function(){
```

Comment:
I'm wondering if there may be a way for us to pull out the javascript and have it be a conditional module within `plugins/openlibrary/js`. I know @jdlrobson has been working really hard to separate js logic from our templates :)

Reply:
Agreed.  This is on my list of todos, but I haven't yet planned it out.

Reply:
Done.  That wasn't as complicated as I thought it would be :)